blades/combo_blade.h

```diff
@@ -0,0 +1,200 @@
+    return real_style_->NoOnOff();
+  }
+
+  virtual bool Charging() { return real_style_->Charging();}
+
+  bool IsHandled(HandledFeature effect) override {
+    if (real_style_)
+      return real_style_->IsHandled(effect);
+    return false;
+  }
+
+  OverDriveColor getColor(int i) override {return real_style_->getColor(i);}
+
+  int get_max_arg(int arg) override { return real_style_->get_max_arg(arg);}
+
+  void setRealStyle(BladeStyle* style){real_style_ = style;}
+
+  BladeStyle* GetRealStyle(){return real_style_;}
+
+  BladeStyle* UnSetRealStyle(){
+    BladeStyle* result = real_style_;
+    real_style_ = nullptr;
+    return result;
+  }
+
+  ComboBladeStyleWrapper(ComboBladeWrapper* comboBlade, BladeBase* primary_blade) :
+    comboBlade_(comboBlade),
+    primary_blade_(primary_blade),
+    real_style_(nullptr){
+  }
+protected:
+  ComboBladeWrapper* comboBlade_;
+  BladeBase* primary_blade_;
+  BladeStyle* real_style_;
+};
+
+/*
+ This class is used by ComboBladeWrapper to allow intercepting the style run()
+ calls sent from the blades inside the ComboBladeWrapper.
+
+This class acts as our entry in the blade array. When activated, it will
+activate the loops of the two assigned blades, pointing them at an instance of
+ComboBladeStyleWrapper. This will then be used to intercept the style run()
+calls of the indivdual blades and instead run the style against the
+ComboBladeWrapper. The ComboBladeWrapper will then send the led set calls
+to the appropriate blade based on the led number.
+
+*/
+class ComboBladeWrapper : public BladeWrapper {
+public:
+  ComboBladeWrapper(BladeBase* blade1, BladeBase* blade2):
+    blade1_num_leds_(blade1->num_leds()),
+    blade2_(blade2)
+  {
+    blade_ = blade1;
+    dummy_style_ = new ComboBladeStyleWrapper(this, blade1);
+  }
+
+  int num_leds() const override {
+    int result = blade1_num_leds_;
+    result += blade2_->num_leds();
+    return result;
+  }
+  void set(int led, Color16 c) override {
+    if (led < blade1_num_leds_) {
+      return blade_->set(led, c);
+    } else {
+      return blade2_->set(led - blade1_num_leds_, c);
+    }
+  }
+  void set_overdrive(int led, Color16 c) override {
+    if (led < blade1_num_leds_) {
+      return blade_->set_overdrive(led, c);
+    } else {
+      return blade2_->set_overdrive(led - blade1_num_leds_, c);
+    }
+  }
+
+  void allow_disable() override {
+    blade_->allow_disable();
+    blade2_->allow_disable();
+  }
+
+  void clear() override {
+    blade_->clear();
+    blade2_->clear();
+  }
+
+  void Activate(int blade_number) override {
+    blade_->Activate(blade_number);
+    blade2_->Activate(blade_number);
+  }
+
+  void Deactivate() override {
+    blade_->Deactivate();
+    blade2_->Deactivate();
+  }
+
+  void SetStyle(BladeStyle* style) override {
+    dummy_style_->setRealStyle(style);
+    blade_->SetStyle(dummy_style_);
+    blade2_->SetStyle(dummy_style_);
+  }
+
+  BladeStyle* UnSetStyle() override {
+    if (dummy_style_)
```

Comment:
If you call blade_ / blade2_->UnsetStyle() here then dummy_style will never have a null style (when used), and we can drop the if() statements in that class.

Reply:
Moving these into a single object resolves this, thanks!